router.js

```diff
@@ -0,0 +1,113 @@
+const Posts = require('./data/db.js');
+
+const express = require('express');
+
+const router = express.Router();
+
+router.get('/', async(req, res) => {
+  try {
+    const posts = await Posts.find(req.query);
+    res.status(200).json(posts);
+  }
+  catch(error) {
+    console.log(error);
+    res.status(500).json({
+      error: "The posts information could not be retreived"
+    })
+  }
+});
+
+router.get('/:id', async(req, res) => {
+  try {
+    const post = await Posts.findById(req.params.id);
+    if(post) {
+      res.status(200).json(post);
+    }
+    else {
+      res.status(404).json({
+        message: "The post with the specifed ID does not exist."
+      })
+    }
+  }
+  catch(error) {
+    console.log(error);
+    res.status(500).json({
+      message: "The post informationc could not be retreived."
+    })
+  }
+})
+
+router.post('/', async(req, res) => {
+  try {
+    const { title, contents, created_at, updated_at } = await Posts.insert(req.body);
+    if(!title || !contents) {
+      res.status(400).json({ errorMessage: "Please provide title and contents for the post."})
+    } else {
```

Comment:
It was throwing this error, but it was, in fact, posting successfully. There's something squirrely about that if statement.